docs/source/development/widget.md

```diff
@@ -251,9 +250,93 @@ See [Storybook](https://6.docs.plone.org/storybook) with available widgets.
 
 ## Write a new widget
 
-```{note}
-Please contribute to this section!
-```
+A custom widget is often used in Volto forms.
+To configure a new Volto widget follow these steps.
+
+1.  Add this import in the first line of {file}`index.js` file of the add-on to import the component that is the actual widget.
+
+    ```jsx
+    import { ToggleWidget } from './ToggleWidget';
+    ```
+
+1.  Add this line before returning from `applyConfig` function inside the {file}`index.js` of the add-on to register the add-on component (widget).
+
+    ```jsx
+    config.widgets.widget.toggle_widget = ToggleWidget;
+    ```
+
+1.  Next, create the widget component file {file}`ToggleWidget.jsx` with these contents:
+
+    ```jsx
+    import { Button } from 'semantic-ui-react';
+
+    export function ToggleWidget({ id, value, onChange }) {
+        return (
+            <Button
+                onClick={(e) => {
+                    e.preventDefault();
+                    onChange(id, value === 'On' ? 'Off' : 'On');
+                }}
+            >
+                Toggle me: {value}
+            </Button>
+        );
+    }
+
+    export default ToggleWidget;
```

Comment:
I ran this through prettier. Code blocks should follow the project's formatting guidance.

```suggestion
    import { Button } from "semantic-ui-react";
    export function ToggleWidget({ id, value, onChange }) {
      return (
        <Button
          onClick={(e) => {
            e.preventDefault();
            onChange(id, value === "On" ? "Off" : "On");
          }}
        >
          Toggle me: {value}
        </Button>
      );
    }

    export default ToggleWidget;
```